### PR TITLE
Improve Resilience Of Signing Checks For MacOS

### LINKED ISSUE #3428: MacOS Signing - Improve Resiliency In sign.sh script
Improve MacOS Signing by improving the resiliency of the signing script. Use similar logic and process to that added to the ci-jenkins-pipelines.

### PR DESCRIPTION
**Pull request by steelhead31**
Fixes #3428 

Include a 20 retry for each library, when it fails to sign at the first attempt.

### CODE REVIEW

File: sign.sh [hunk 1]
```diff
@@ -132,11 +136,49 @@ signRelease()
...
+                success=false
+                iteration=$((iteration+1))
+                errcount=$((errcount+1))
+              fi
+            done
+            if [ "$errcount" -gt 0 ]
+            then
+              echo "Errors Encountered During Signing"
+              echo "Error Count = $errcount"
+              exit 1
```
Comment (adamfarley) [inline sign.sh:176]: I'm not sure I understand. I thought this fix was intended to retry 20 times in order to *avoid* failures. This appears to say "If we needed to retry downloading this file even once, then fail".

How would this be better than the current situation?

**Review by adamfarley — CHANGES_REQUESTED** [review]
Confused. Your changes appear to cancel themselves out. Thoughts please.

File: sign.sh (hunk 1 above)
Comment (steelhead31) [reply sign.sh:176]: The multi level nested loops and if statements make this slightly difficult to read.

Comment (steelhead31) [reply sign.sh]: Looks like its working ok in a successful run, Ive also tested it in my local test script, and it works correctly,, with a failure followed by a success

Comment (adamfarley) [reply sign.sh]: Thanks Scott. Looks excellent, well done. :)

**Review by adamfarley — APPROVED** [review]
LGTM 👍

**Review by andrew-m-leonard — APPROVED** [review]
lgtm

**Pull request merged**